proc_echo_mw.py

```diff
@@ -1,143 +1,101 @@
+    #slice out the frequency range along t2 axis
     s = s['t2':freq_range]
+    
+    #inverse fourier transform into time domain
     s.ift('t2')
-    residual,best_shift = hermitian_function_test(s[
-        'ph2',-2]['ph1',1])
-    fl.next('hermitian test')
-    fl.plot(residual)
-    s.setaxis('t2',lambda x: x-best_shift)
-    s.register_axis({'t2':0}, nearest=False)
-    # {{{ implement zeroth-order correction
-    # note that it's not going to have only one
-    # dimension, b/c we will have at least a power
-    # dimension
-    ph0 = s['t2':0]['ph2',-2]['ph1',1]
-    s /= zeroth_order_ph(ph0, fl=fl)
-    if s['t2':0]['ph2',-2]['ph1',1]['power',0].real < 0:
-        s *= -1
-    # }}}
+
+    #visualize time domain after filtering and phasing 
+    logger.info(strm("THIS IS THE SHAPE"))
+    logger.info(strm(ndshape(s)))
+    s = slice_FID_from_echo(s)['t2':(None,0.05)]
+    fl.side_by_side('time domain (after filtering and phasing)\n$\\rightarrow$ use to adjust time range', s, time_range)
+    
+    #slices out time range along t2 axis
+    s =s['t2':time_range]
+
+    #visualize centered
+    echo_start = s.getaxis('t2')[0]
+    dw = diff(s.getaxis('t2')[r_[0,1]]).item()
+    centered_echo = s['t2':(echo_start,-echo_start+dw)]
+    plotdata = abs(centered_echo/centered_echo['t2',::-1])
+    fl.next('plot data')
+    plotdata[lambda x: x>2] = 2
+    fl.image(plotdata)
+    
+    #slice out the FID from the echoes
+    fl.next('FID slice')
+    logger.info(strm("THIS IS THE SHAPE"))
+    logger.info(strm(ndshape(s)))
+    s = slice_FID_from_echo(s)['t2':(None,0.05)]
+    
+    #redefine time range along t2
     fl.side_by_side('time domain (after filtering and phasing)\n$\\rightarrow$ use to adjust time range',
-            s,time_range)
+        s,time_range)
     s = s['t2':time_range]
-    # {{{ all of this is to check and see if we think
-    # we can add the two sides of the echo to increase
-    # our SNR -- right now, it doesn't look like it
+    
+    #mirror test to test centered data
     fl.next('echo mirror test')
     echo_start = s.getaxis('t2')[0]
     dw = diff(s.getaxis('t2')[r_[0,1]]).item()
     centered_echo = s['t2':(echo_start,-echo_start+dw)]
     plotdata = abs(centered_echo/centered_echo['t2',::-1])
+    fl.next('plot data')
     plotdata[lambda x: x>2] = 2
     fl.image(plotdata)
-    # }}}
+    
+    #apodizing and zero fill
     fl.next('apodize and zero fill')
     R = 5.0/(time_range[-1]) # assume full decay by end time
     s *= exp(-s.fromaxis('t2')*R)
     s.ft('t2',pad=1024)
     fl.image(s)
-    # {{{ select FID
+    
+    #select coherence channel in time domain
     s.ift('t2')
     s = s['ph2',-2]['ph1',1]['t2':(0,None)]
     s['t2',0] *= 0.5
     s.ft('t2')
-    # }}}
+    
+    #plotting enhancement curve at lowest and highest powers 
     fl.next('compare highest power to no power')
     idx_maxpower = argmax(s.getaxis('power'))
     fl.plot(s['power',0])
     fl.plot(s['power',idx_maxpower])
```

Comment:
for future reference, note that in vim, if using my vimrc, which I think you are, you can use shift-V to highlight a block of code, and then type BLO (for block) -- this will give you the `#{{{` and `# }}}` markers -- these are nice, because they fold in vim, leaving you with just a description of the code, which allows you to see an outline of what the code does

Reply:
okay done